pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -263,14 +278,14 @@ def is_optimized_sync(repository, remote, url):
 
     """
     with tempfile.TemporaryDirectory("."):
-        result = get_repomd_file(remote, url)
-        if not result:
+        try:
+            result = get_repomd_file(remote, url)
+            repomd_path = result.path
+            repomd = cr.Repomd(repomd_path)
```

Comment:
Handing Repomd() a bugus file that we thought was a repomd.xml was what opened this can of worms in the first place. Could we be eating a possible exception here, that would be telling us someting?

Reply:
I don't think so, it's hardcoded that this is only going to return a file under the path `repodata/repomd.xml`.  Am I missing something?

Reply:
Hm, no I think you're right. The only reason that failed us, was because of the mirrorlist-bug that I addressed last week.